builder/metadata.go

```diff
@@ -187,26 +197,32 @@ func (b *Builder) UpdateVersions(nextMix, nextUpstream uint32) error {
 
 	nextUpstreamStr := strconv.FormatUint(uint64(nextUpstream), 10)
 
-	nextFormat := format
-	if nextUpstream > latest {
-		nextFormat, err = b.getUpstreamFormat(nextUpstreamStr)
-		if err != nil {
-			return err
+	var nextFormat string
+	if checkUpstream {
+		nextFormat = format
+		if nextUpstream > latest {
+			nextFormat, err = b.getUpstreamFormat(nextUpstreamStr)
+			if err != nil {
+				return err
+			}
 		}
-	}
 
-	// Verify the version exists by checking if its Manifest.MoM is around.
-	_, err = b.DownloadFileFromUpstreamAsString(fmt.Sprintf("/update/%d/Manifest.MoM", nextUpstream))
-	if err != nil {
-		return errors.Wrapf(err, "invalid upstream version %d", nextUpstream)
+		// Verify the version exists by checking if its Manifest.MoM is around.
+		_, err = b.DownloadFileFromUpstreamAsString(fmt.Sprintf("/update/%d/Manifest.MoM", nextUpstream))
+		if err != nil {
+			return errors.Wrapf(err, "invalid upstream version %d", nextUpstream)
+		}
 	}
 
 	fmt.Printf(`Old mix:      %d
 Old upstream: %d (format: %s)
 
-New mix:      %d
-New upstream: %d (format: %s)
-`, b.MixVerUint32, b.UpstreamVerUint32, format, nextMix, nextUpstream, nextFormat)
+New mix:      %d\n
```

Comment:
Why did you decide to add an extra newline here?

This should really be a template instead of a `Printf` like this, but I guess that's out of scope for what you are trying to do.

Reply:
It's the same output as before, the original was 
```
`
%s
`
```

which is the same as
`"%s\n"` because of the backticks.
yeah more templates more better at some point.